src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -23,39 +24,53 @@
             + "Example: " + COMMAND_WORD + " 1";
 
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person(s):\n%1$s";
 
-    private final Index targetIndex;
+    private ArrayList<Index> targetIndexArraylist;
+
+    public DeleteCommand(ArrayList<Index> targetIndexArraylist) {
+        this.targetIndexArraylist = targetIndexArraylist;
+    }
 
     public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+        ArrayList<Index> targetIndexArrayList = new ArrayList<Index>();
+        targetIndexArrayList.add(targetIndex);
+        this.targetIndexArraylist = targetIndexArraylist;
     }
 
 
     @Override
     public CommandResult executeUndoableCommand() throws CommandException {
 
         List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+        String result = "";
+        String people = "";
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        for (Index i : targetIndexArraylist) {
+            if (i.getZeroBased() >= lastShownList.size()) {
```

Comment:
There is a limitation as to placing this check within the for-loop. Let's say AddressBook is of size 10 and we run the following command:
`delete 2 3 4 11 5 6 7`
Persons at index 2, 3, and 4 will be deleted, but 5, 6, and 7 will not because the method halts by the time it checks that index 11 is beyond AddressBook size.

This is not wrong, as it depends on your desired design. But I implemented a way to fix this. You can take a read at this class in method `executeUndoableCommand()` where I added block comments:
https://github.com/CS2103AUG2017-W10-B2/main/blob/1b592d12a28c72b3288570f44de4484e2b9f172a/src/main/java/seedu/address/logic/commands/FavoriteCommand.java#L47